### PR TITLE
fix: Grid new row getting cut

### PR DESCRIPTION
**Pull request by Ahad-patel**
closes : #6330 

### Feature Preview

I have replaced stack with column in _renderList, as when the CalculationsRow are moved from the viewport the _PositionedCalculationsRow are stacked on top of the ReorderableListView which was hiding the text rows below, so I moved them in column and removed position widget from both ReorderableListView and _PositionedCalculationsRow and expanded the ReorderableListView so it aquired all the availiable space of the window

https://github.com/user-attachments/assets/5fe04031-da28-414a-b264-39e830e37291

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by Ahad-patel** [comment]
Hey @LucasXu0, I have added a test case and fixed the row getting cut. Please let me know if any more changes needed

**Comment by Xazin** [comment]
Is there a reason for not just insetting the positioned bottom with the height of the calculations row as a height? Or did I misunderstand the issue?

**Review by Xazin — APPROVED** [review]

File: frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart [hunk 1]
```diff
@@ -358,9 +358,9 @@ class _GridRowsState extends State<_GridRows> {
     // 2. GridCalculationsRow
     // 3. Footer Padding
     final itemCount = state.rowInfos.length + 3;
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart:360]:
```suggestion
    final itemCount = state.rowInfos.length + 2;
```

File: frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart [hunk 2]
```diff
@@ -422,9 +422,8 @@ class _GridRowsState extends State<_GridRows> {
 
               if (index == state.rowInfos.length + 1) {
                 if (showFloatingCalculations) {
-                  return const SizedBox(
+                  return const SizedBox.shrink(
                     key: Key('calculations_bottom_padding'),
-                    height: 36,
                   );
```
[the remark this answers is not in the document]
Comment (Ahad-patel) [reply frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart:427]: ok, I'll fix that

[the remark this answers is not in the document]
Comment (Ahad-patel) [reply frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart:427]: https://github.com/user-attachments/assets/9f588e48-cfdb-4183-b7f7-0d8c5e42b27a

As you suggested I removed the `showFloatingCalculations` if condition, But on doing that the `GridCalculationsRow` will always be visible, so on scrolling to end calculation row will be visible twice

[the remark this answers is not in the document]
Comment (Xazin) [reply frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart:427]: Like this:
```
              if (index == state.rowInfos.length + 1 &&
                  !showFloatingCalculations) {
                return GridCalculationsRow(
                  key: const Key('grid_calculations'),
                  viewId: widget.viewId,
                );
              }
```

[the remark this answers is not in the document]
Comment (Xazin) [reply frontend/appflowy_flutter/lib/plugins/database/grid/presentation/grid_page.dart:427]: Let me push a commit to clean up some of the code regarding this and implement the above :+1:

**Pull request merged**